compaction/compaction_manager.cc

```diff
@@ -1520,6 +1524,11 @@ future<> compaction_manager::perform_cleanup(owned_ranges_ptr sorted_owned_range
             };
             filter_candidates(t.main_sstable_set().all());
             filter_candidates(t.maintenance_sstable_set().all());
+
+            if (!t.cleanup_sstable_set().empty()) {
+                t.set_owned_ranges_ptr(sorted_owned_ranges);
+            }
```

Comment:
Why is this conditional?

Shouldn't we set it during table_state creation (and update when changed)?

btw, it should be trimmed to the compaction_group's range (I guess in table.cc)

Reply:
1. in next version it will be in compaction_state, not table_state
2. currently it also indicates that cleanup is required in the compaction group, so if it's set unconditionally, we'd need another way to trigger the slightly less efficient cleanup-related logic.

Reply:
> btw, it should be trimmed to the compaction_group's range (I guess in table.cc)

Right.  It's a preexisting issue.  I think it's harmless with static compaction groups but nevertheless it should be fixed.

Cc @raphaelsc

Reply:
I see you already worked on that. Will review the series.